### PR TITLE
Simplify `set_os_env_cache()` cache to work on Posit Workbench

### LINKED ISSUE #706: Unable to install paws.common 0.6.3 from CRAN
Thank you for the recent 0.6.3 release! Unfortunately, I'm having issues installing from CRAN (running on Workbench hosted on Ubuntu jammy):

Installing from `https://cran.rstudio.com/`:
```
> install.packages('paws.common')
Installing package into ‘/home/michael.thomson/R/x86_64-pc-linux-gnu-library/4.1’
(as ‘lib’ is unspecified)
trying URL 'https://cran.rstudio.com/src/contrib/paws.common_0.6.3.tar.gz'
Content type 'application/x-gzip' length 112727 bytes (110 KB)
==================================================
downloaded 110 KB

* installing *source* package ‘paws.common’ ...
** package ‘paws.common’ successfully unpacked and MD5 sums checked
** using staged installation
** libs
g++ -std=gnu++14 -I"/opt/R/4.1.3/lib/R/include" -DNDEBUG  -I'/home/michael.thomson/R/x86_64-pc-linux-gnu-library/4.1/Rcpp/include' -I/usr/local/include   -fpic  -g -O2  -c RcppExports.cpp -o RcppExports.o
g++ -std=gnu++14 -I"/opt/R/4.1.3/lib/R/include" -DNDEBUG  -I'/home/michael.thomson/R/x86_64-pc-linux-gnu-library/4.1/Rcpp/include' -I/usr/local/include   -fpic  -g -O2  -c encoding.cpp -o encoding.o
g++ -std=gnu++14 -shared -L/opt/R/4.1.3/lib/R/lib -L/usr/local/lib -o paws.common.so RcppExports.o encoding.o -L/opt/R/4.1.3/lib/R/lib -lR
installing to /home/michael.thomson/R/x86_64-pc-linux-gnu-library/4.1/00LOCK-paws.common/00new/paws.common/libs
** R
** byte-compile and prepare package for lazy loading
Loading required package: usethis
** help
*** installing help indices
** building package indices
Loading required package: usethis
** testing if installed package can be loaded from temporary location
Loading required package: usethis
Error: package or namespace load failed for ‘paws.common’:
 .onLoad failed in loadNamespace() for 'paws.common', details:
  call: `[<-`(`*tmp*`, found, 2, value = "")
  error: (subscript) logical subscript too long
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/home/michael.thomson/R/x86_64-pc-linux-gnu-library/4.1/paws.common’
Warning in install.packages :
  installation of package ‘paws.common’ had non-zero exit status

The downloaded source packages are in
	‘/tmp/RtmpjrTKL0/downloaded_packages’
```

Similarly, pulling from 54db5e3f10a1fae0eea56a3c3207ea464e4ab78e :
```
> remotes::install_github("paws-r/paws/paws.common", ref = "54db5e3f10a1fae0eea56a3c3207ea464e4ab78e")
Downloading GitHub repo paws-r/paws@54db5e3f10a1fae0eea56a3c3207ea464e4ab78e
'/usr/bin/git' clone --depth 1 --no-hardlinks --recurse-submodules https://github.com/aws/aws-sdk-js /tmp/RtmpjrTKL0/remotes11b33c47acbd/paws-r-paws-54db5e3/paws.common/../vendor/aws-sdk-js
Cloning into '/tmp/RtmpjrTKL0/remotes11b33c47acbd/paws-r-paws-54db5e3/paws.common/../vendor/aws-sdk-js'...
── R CMD build ──────────────────────────────────────────────────────────────────────────────────────────────────────────────
✔  checking for file ‘/tmp/RtmpjrTKL0/remotes11b33c47acbd/paws-r-paws-54db5e3/paws.common/DESCRIPTION’ (346ms)
─  preparing ‘paws.common’:
✔  checking DESCRIPTION meta-information ...
─  cleaning src
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘paws.common_0.6.3.tar.gz’
   Warning in sprintf(gettext(fmt, domain = domain), ...) :
     one argument not used by format 'invalid uid value replaced by that for user 'nobody''
   Warning: invalid uid value replaced by that for user 'nobody'
   Warning in sprintf(gettext(fmt, domain = domain), ...) :
     one argument not used by format 'invalid gid value replaced by that for user 'nobody''
   Warning: invalid gid value replaced by that for user 'nobody'
   
Installing package into ‘/home/michael.thomson/R/x86_64-pc-linux-gnu-library/4.1’
(as ‘lib’ is unspecified)
* installing *source* package ‘paws.common’ ...
** using staged installation
** libs
g++ -std=gnu++14 -I"/opt/R/4.1.3/lib/R/include" -DNDEBUG  -I'/home/michael.thomson/R/x86_64-pc-linux-gnu-library/4.1/Rcpp/include' -I/usr/local/include   -fpic  -g -O2  -c RcppExports.cpp -o RcppExports.o
g++ -std=gnu++14 -I"/opt/R/4.1.3/lib/R/include" -DNDEBUG  -I'/home/michael.thomson/R/x86_64-pc-linux-gnu-library/4.1/Rcpp/include' -I/usr/local/include   -fpic  -g -O2  -c encoding.cpp -o encoding.o
g++ -std=gnu++14 -shared -L/opt/R/4.1.3/lib/R/lib -L/usr/local/lib -o paws.common.so RcppExports.o encoding.o -L/opt/R/4.1.3/lib/R/lib -lR
installing to /home/michael.thomson/R/x86_64-pc-linux-gnu-library/4.1/00LOCK-paws.common/00new/paws.common/libs
** R
** byte-compile and prepare package for lazy loading
Loading required package: usethis
** help
*** installing help indices
** building package indices
Loading required package: usethis
** testing if installed package can be loaded from temporary location
Loading required package: usethis
Error: package or namespace load failed for ‘paws.common’:
 .onLoad failed in loadNamespace() for 'paws.common', details:
  call: `[<-`(`*tmp*`, found, 2, value = "")
  error: (subscript) logical subscript too long
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/home/michael.thomson/R/x86_64-pc-linux-gnu-library/4.1/paws.common’
Warning message:
In i.p(...) :
  installation of package ‘/tmp/RtmpjrTKL0/file11b3240e7aaf/paws.common_0.6.3.tar.gz’ had non-zero exit status
```

### PR DESCRIPTION
**Pull request by fh-mthomson**
Closes #706 

Introduced in https://github.com/paws-r/paws/pull/693

### CODE REVIEW

**Comment by fh-mthomson** [comment]
@DyfanJones I don't know enough about the use case for this cache functionality to test in depth, but I was able to confirm:

1. On both Mac and Posit Workbench (jammy image), I could (a) load the package from this branch and (b) install via `devtools::install_github("fh-mthomson/paws/paws.common@mthomson/os_env_vars")`
2. Cache in `package:paws.common` looks ~similar across branches

Before (Mac):
image

After (Mac): 
image

File: paws.common/tests/testthat/test_cache.R [hunk 1]
```diff
@@ -30,17 +30,17 @@ test_that("check if environmental variables are parsed correctly", {
     paste(sample(letters, 10), collapse = ""),
```
Comment (fh-mthomson) [inline paws.common/tests/testthat/test_cache.R:30]:
```
> devtools::test(filter = "cache")
ℹ Testing paws.common
✔ | F W  S  OK | Context
✔ |          6 | cache                                                                                                           

══ Results ══════════════════════════════════════════════════════════════════════════════════════════════════════════════════════
[ FAIL 0 | WARN 0 | SKIP 0 | PASS 6 ]
````

File: paws.common/R/cache.R [hunk 2]
```diff
@@ -3,17 +3,19 @@ ini_cache <- new.env(parent = emptyenv())
 os_env_cache <- new.env(parent = emptyenv())
 
 set_os_env_cache <- function() {
-  env <- system("printenv", intern = TRUE)
+  env <- Sys.getenv()
+
+  # only cache AWS_* environment variables to avoid caching sensitive information
```
Comment (fh-mthomson) [inline paws.common/R/cache.R:8]: lmk if you disagree with this! personally, I felt a bit uneasy that _all_ env vars would be cached upon load without input from this user, so this felt better scoped to only the env vars relevant to this package:

image

File: paws.common/R/RcppExports.R [hunk 3]
```diff
@@ -4,6 +4,5 @@
 #' @useDynLib paws.common _paws_common_paws_url_encoder
 #' @importFrom Rcpp evalCpp
 paws_url_encoder <- function(urls, safe = "") {
-    .Call('_paws_common_paws_url_encoder', PACKAGE = 'paws.common', urls, safe)
+  .Call("_paws_common_paws_url_encoder", PACKAGE = "paws.common", urls, safe)
```
Comment (fh-mthomson) [inline paws.common/R/RcppExports.R:7]: ran `styler::style_dir()` while I was here :)

**Comment by fh-mthomson** [comment]
Confirmed! Adding `AWS_TEST` to my `~/.Renvirion` then reloading:
```
> library(paws.common)
> 
> paws.common:::os_env_cache |> as.list()
$AWS_TEST
[1] "yes"
```
Then, if I run a command to initialize temporary AWS creds, I see these as well:
```
> detach("package:paws.storage", unload = TRUE)
> detach("package:paws.common", unload = TRUE)
> library(paws.common)
> paws.common:::os_env_cache |> as.list() %>% names()
[1] "AWS_CREDENTIAL_EXPIRATION" "AWS_TEST"                  "AWS_ACCESS_KEY_ID"         "AWS_SESSION_TOKEN"        
[5] "AWS_SECRET_ACCESS_KEY"  
```

**Comment by DyfanJones** [comment]
I will release this in the next couple of days as it is stopping people from installing `paws.common` on linux (which is not good :P)

**Comment by fh-mthomson** [comment]
Thank you for your quick resolution, as always @DyfanJones !

**Pull request merged**